etc/kayobe/ansible/scripts/test_smartmon.py

```diff
@@ -0,0 +1,262 @@
+                        line.endswith(" 1")
+                        for line in metrics
+                    )
+                    self.assertTrue(
+                        smart_available_found,
+                        f"Expected device_smart_available=1 for {dev_name}, not found."
+                    )
+
+                # If smart_enabled is true, we expect device_smart_enabled = 1
+                if device_info.get("smart_enabled"):
+                    smart_enabled_found = any(
+                        line.startswith("device_smart_enabled{") and
+                        f'disk="{dev_name}"' in line and
+                        line.endswith(" 1")
+                        for line in metrics
+                    )
+                    self.assertTrue(
+                        smart_enabled_found,
+                        f"Expected device_smart_enabled=1 for {dev_name}, not found."
+                    )
+
+                # device_smart_healthy if assessment in [PASS, WARN, FAIL]
+                # PASS => 1, otherwise => 0
+                assessment = device_info.get("assessment", "").upper()
+                if assessment in ["PASS", "WARN", "FAIL"]:
+                    expected_val = 1 if assessment == "PASS" else 0
+                    smart_healthy_found = any(
+                        line.startswith("device_smart_healthy{") and
+                        f'disk="{dev_name}"' in line and
+                        line.endswith(f" {expected_val}")
+                        for line in metrics
+                    )
+                    self.assertTrue(
+                        smart_healthy_found,
+                        f"Expected device_smart_healthy={expected_val} for {dev_name}, not found."
+                    )
+
+    def test_parse_if_attributes(self):
+        """
+        Test parse_if_attributes() for every JSON fixture in ./drives/.
+        We do subTest() so each fixture is tested individually.
+        """
+        for fixture_path in self.fixture_files:
+            fixture_name = os.path.basename(fixture_path)
+            with self.subTest(msg=f"Testing if_attributes with {fixture_name}"):
+                data = load_json_fixture(fixture_name)
+                device_info = data["device_info"]
+                if_attrs = data.get("if_attributes", {})
+
+                device = self.create_mock_device_from_json(device_info, if_attrs)
+                metrics = parse_if_attributes(device)
+
+                dev_name = device_info["name"]
+                dev_iface = device_info["interface"]
+                dev_serial = device_info["serial"].lower()
+
+                # For each numeric attribute in JSON, if it's in SMARTMON_ATTRS,
+                # we expect a line in the script's output.
+                for attr_key, attr_val in if_attrs.items():
+                    # Convert from e.g. "criticalWarning" -> "critical_warning"
+                    snake_key = re.sub(r'(?<!^)(?=[A-Z])', '_', attr_key).lower()
```

Comment:
I'd consider this too complex for a unit test. It would be better to hard code the expected outcomes for a given input.

Reply:
Should we do this as a part 2? I think it would be a lot of work to hard code it